databroker/broker.py

```diff
@@ -575,6 +545,68 @@ def get_events(self, header, pvs):
                 yield Document('Event', doc)
 
 
+class DangerBroker(Broker):
+    def get_resource_uids(self, header):
+        '''Given a Header, give back a list of resource uids
+
+        These uids are required to move the underlying files.
+
+        Parameters
+        ----------
+        header : Header
+
+        Returns
+        -------
+        ret : set
+            set of resource uids which are refereneced by this
+            header.
+        '''
+        external_keys = set()
+        for d in header['descriptors']:
+            for k, v in d['data_keys'].items():
+                if 'external' in v:
+                    external_keys.add(k)
+
+        ev_gen = self.get_events(header, stream_name=None,
+                                 fields=external_keys, fill=False)
+        resources = set()
+        for ev in ev_gen:
+            for v in ev['data'].values():
+                res = self.fs.resource_given_eid(v)
```

Comment:
what's the different between ...given_eid and ...given_uid? eid is for individual datum?